src/Evolution/Systems/GrMhd/ValenciaDivClean/KastaunEtAl.cpp

```diff
@@ -250,11 +250,22 @@ Primitives FunctionOfMu<ThermodynamicDim>::primitives(const double mu) const {
       q_ - 0.5 * b_squared_ -
       0.5 * square(mu * x) * (r_squared_ * b_squared_ - r_dot_b_squared_);
   // Equation (42) with bounds from Equation (6)
-  const double epsilon_hat = std::clamp(
-      w_hat * (q_bar - mu * r_bar_squared) +
-          v_hat_squared * square(w_hat) / (1.0 + w_hat),
-      equation_of_state_.specific_internal_energy_lower_bound(rho_hat),
-      equation_of_state_.specific_internal_energy_upper_bound(rho_hat));
+  double epsilon_hat{};
+  if constexpr (ThermodynamicDim == 3) {
+    epsilon_hat =
+        std::clamp(w_hat * (q_bar - mu * r_bar_squared) +
+                       v_hat_squared * square(w_hat) / (1.0 + w_hat),
+                   equation_of_state_.specific_internal_energy_lower_bound(
+                       rho_hat, electron_fraction_),
+                   equation_of_state_.specific_internal_energy_upper_bound(
+                       rho_hat, electron_fraction_));
```

Comment:
Evaluate only the EOS bounds in the `if` and then compute epsilon outside the `if` to make clear that the expression for epsilon is independent of the thermodynamic dim.

Reply:
No, that spills scope and adds extra complexity.

I would do:
```cpp
double epsilon_hat = w_hat * (q_bar - mu * r_bar_squared) +
                       v_hat_squared * square(w_hat) / (1.0 + w_hat);
if constexpr ((ThermodynamicDim == 3) {
    epsilon_hat =
        std::clamp(epsilon_hat,
                   equation_of_state_.specific_internal_energy_lower_bound(
                       rho_hat, electron_fraction_),
                   equation_of_state_.specific_internal_energy_upper_bound(
                       rho_hat, electron_fraction_));
} else {
    epsilon_hat = std::clamp(
       epsilon_hat,
        equation_of_state_.specific_internal_energy_lower_bound(rho_hat),
        equation_of_state_.specific_internal_energy_upper_bound(rho_hat));
  }

```